### DISCUSSION TITLE
Overall Turnaround MS

### DISCUSSION THREAD

**Discussion by AndyMcProducer** [jamulussoftware/jamulus#969] [Technical Support & Help] [answered]
Hiya, what shoudl or what can the overall ms be when connected to local host?
Even at 32buffers I'm getting 16 to 20ms.

Anyone have some info or tweaks to share on this?

**Comment by pljones** [comment]
The best place to start: https://jamulus.io/wiki/Getting-Started

**Comment by AndyMcProducer** [comment]
Ping to local host is 0 to 1ms but turnaround is 16ms. So this pushes towards 40ms and over sometimes.

**Reply by jujudusud** [reply]
It is not a problem if you have 40 ms of total latency. Use your ears instead of the Jamulus latency display.

**Comment by ann0see** [comment]
Which OS are you using? Do you use an audio interface? If yes which one?

**Comment by gene96817** [comment]
I am seeing pings 0-3 ms and an overall delay of 10-13ms (with the shortest buffer).

**Comment by gene96817** [comment]
for my case, MacOSss 11.1, built-in audio, USB mic (Blue Snowball).

**Comment by pljones** [comment]
From my upstairs machine, through one switch to the main switch downstairs, then back to the switch where my server(s) live, I get 0ms ping, 10-12ms overall delay.  About the same downstairs (that only has the one switch to go through).  64 samples per buffer on both.  Echo AudioFire4 upstairs, Zoom UAC-2 downstairs.

**Comment by dcorson-ticino-com** [comment, marked as answer]
When the ping time is 0, the overall delay is the time your interface introduces (more or less).  Jitter buffering adds time too, but with ping 0 there will be little or no jitter so that can be ruled out.
There are many audio interfaces that have latencies of 10-12ms, but I have also seen interfaces with much more such as internal audio interfaces.

**Comment by AndyMcProducer** [comment]
I'll try running the server on another machine on the lan here. I got 2 sound devices and both seem to hit around the same, if i turn off auto on the jitter then o can get it down but not tested the sound with it down yet.

**Comment by digitblues** [comment]
DonC is right.  If your server is local host with 0 ping time then the latency is introduced by the interface and jitter buffer.  When I first learned to use Jamulus, I was frustrated by the high latency value and had tried to find ways to reduce it by trial and error on different OS and interface/drivers.  I tested it on Win7, Win10 and Ubuntu using Behringer, Focusrite and ASIO4ALL.  I came to the conclusion that the interface will introduce at least 15ms.  The best OS to achieve this low value is Ubuntu pairing with Behringer UMC404HD interface.  Focusrite 18i20 Gen 3 is higher about 18-20 ms.  The latency is further increased when moving to Windows platform.  With ASIO4ALL as a driver to connect to the computer built-in audio port the latency will be the worst with at least 36ms.  One can imagine how much more delay to add on top of this basic latency when the network is include.  Hope this is useful.

**Reply by jujudusud** [reply]
The best way is to use your ears instead of looking on how much latency Jamulus shows. ;-)

**Reply by dcorson-ticino-com** [reply]
The Windows times you mention are not caused by Win10, but by the ASIO4ALL driver.
I run Win10 with a NI Komplete Audio 2 interface and it's native ASIO driver and have overall delays of 12ms more than the ping time.  I don't think it would be better on any of the other OS's.
To get good delays under Win10 you need an interface with a native ASIO driver.  ASIO4ALL adds way too much latency.

**Comment by jujudusud** [comment]
> Hiya, what shoudl or what can the overall ms be when connected to local host?
> Even at 32buffers I'm getting 16 to 20ms.
> 
> Anyone have some info or tweaks to share on this?

You should use 128 "buffers" in order to reduced at its maximum the Jitter buffer of Jamulus client and server.

**Reply by dcorson-ticino-com** [reply]
> You should use 128 "buffers" in order to reduced at its maximum the Jitter buffer of Jamulus client and server.

Note that, at least under Win10, if the connection quality and computer speed allow, using 64 sample buffers and also the small buffer enable can significantly reduce the delay times.  As I have understood from this forum, this seems not to be the case for Linux.
Anyway, just try all the possibilities and see what works best for you.
